### PR TITLE
Put mod_private opts in a map with defaults and updating tests

### PR DESCRIPTION
**Pull request by JanuszJakubiec**

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/107063) / small_tests / d234b2b0295d6e60af28a1698826221d10388e42
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3549/107063/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3549/107063/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-17_10.27.49/lib.mongooseim.logs/run.2022-02-17_10.27.50/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/107084) / small_tests / 8e9e97b7a8a168d520631de73c7ffd6d9fb5d63f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3549/107084/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3549/107084/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-17_10.35.22/lib.mongooseim.logs/run.2022-02-17_10.35.22/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/107131) / small_tests / e8217359e2e739a7c39933341add070336697f31
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3549/107131/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3549/107131/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-17_10.44.09/lib.mongooseim.logs/run.2022-02-17_10.44.09/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/107284) / small_tests / 62c63ef4f382b574e113ff110c15b7b194fbe9eb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3549/107284/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3549/107284/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-17_12.31.37/lib.mongooseim.logs/run.2022-02-17_12.31.38/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/107303) / small_tests / 57cb99f6332c5e4e33533a55c7fbb8bc8b11c952
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3549/107303/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3549/107303/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-17_13.52.06/lib.mongooseim.logs/run.2022-02-17_13.52.07/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/107396) / small_tests / ed08757e065e07532fa93fd2c7930cd1f5d505f3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3549/107396/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3549/107396/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-02-17_14.41.05/lib.mongooseim.logs/run.2022-02-17_14.41.06/suite.log.html)

File: src/mod_private.erl [hunk 1]
```diff
@@ -69,16 +69,19 @@ get_personal_data(Acc, HostType, #jid{ luser = LUser, lserver = LServer }) ->
 %% ------------------------------------------------------------------
 %% gen_mod callbacks
 
+-spec start(HostType :: mongooseim:host_type(), Opts :: gen_mod:module_opts()) -> ok.
```
Comment (gustawlippa) [inline src/mod_private.erl:72]: Isn't this actually `-> ok | {error, atom()}`?

File: src/mod_private.erl [hunk 2]
```diff
@@ -69,16 +69,19 @@ get_personal_data(Acc, HostType, #jid{ luser = LUser, lserver = LServer }) ->
...
 start(HostType, Opts) ->
-    mod_private_backend:init(HostType, Opts),
     IQDisc = gen_mod:get_opt(iqdisc, Opts, one_queue),
+    mod_private_backend:init(HostType, Opts),
```
Comment (gustawlippa) [inline src/mod_private.erl:75]: Why this change actually? 😄

File: src/mod_private.erl [hunk 3]
```diff
@@ -69,16 +69,19 @@ get_personal_data(Acc, HostType, #jid{ luser = LUser, lserver = LServer }) ->
...
     ejabberd_hooks:add(hooks(HostType)),
     gen_iq_handler:add_iq_handler_for_domain(HostType, ?NS_PRIVATE, ejabberd_sm,
                                              fun ?MODULE:process_iq/5, #{}, IQDisc).
 
+-spec stop(HostType :: mongooseim:host_type()) -> ok.
 stop(HostType) ->
     ejabberd_hooks:delete(hooks(HostType)),
-    gen_iq_handler:remove_iq_handler_for_domain(HostType, ?NS_PRIVATE, ejabberd_sm).
+    gen_iq_handler:remove_iq_handler_for_domain(HostType, ?NS_PRIVATE, ejabberd_sm),
+    ok.
```
Comment (gustawlippa) [inline src/mod_private.erl:84]: I'd rather leave the result of `remove_iq_handler_for_domain` and change the spec to `-> ok | {error, not_registered}`. gen_mod's stop callback can return `any()` from what I see. Is there a reason to return a `ok` here?

Comment (gustawlippa) [reply src/mod_private.erl:84]: It's not like it's a big issue, as the result is discarded anyway, but still...

**Review by chrzaszcz — COMMENTED** [review]
I added a few comments.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/107683) / small_tests / 5e284dc9f6970c555f3ae95e51f3f191c0c39fe7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3549/107683/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3549/107683/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-18_13.37.33/lib.mongooseim.logs/run.2022-02-18_13.37.34/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/107755) / small_tests / 246a677de9db3ebbf2139652748029a2920307cc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3549/107755/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3549/107755/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-02-18_17.42.45/lib.mongooseim.logs/run.2022-02-18_17.42.45/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍
